snuba/query/parser/validation/functions.py

```diff
@@ -54,23 +85,16 @@ def validate(self, exp: Expression, data_source: DataSource) -> None:
                     common_function_validators,
                     exc_info=True,
                 )
-            data_model = entity.get_data_model()
-        elif isinstance(data_source, (Query, JoinClause)):
-            # TODO: This is currently ignoring entity validators. We should be validating
-            # each entity.
-            data_model = data_source.get_columns()
-        else:
-            return
 
-        validators = ChainMap(default_validators, entity_validators)
-        try:
-            # TODO: Decide whether these validators should exist at the Dataset or Entity level
-            validator = validators.get(exp.function_name)
-            if validator is not None:
-                validator.validate(exp.parameters, data_model)
-        except InvalidFunctionCall as exception:
-            raise InvalidExpressionException(
-                exp,
-                f"Illegal call to function {exp.function_name}: {str(exception)}",
-                report=False,
-            ) from exception
+            validators = ChainMap(default_validators, entity_validators)
+            try:
+                # TODO: Decide whether these validators should exist at the Dataset or Entity level
+                validator = validators.get(exp.function_name)
```

Comment:
IF you end up applying my suggestion above, this one would have to change a bit as the `AllowedFunctionValidator` would have to be applied to all function calls.